crates/wallet/src/wallet/params.rs

```diff
@@ -149,8 +149,8 @@ impl LoadParams {
         self
     }
 
-    /// Checks that `descriptor` of `keychain` matches this, and extracts private keys (if
-    /// available).
+    /// Checks that the given descriptors match those loaded from persistence and extracts private
+    /// keys if available.
```

Comment:
I have an idea to make the API more understandable (hopefully). We can rename this method to `check_descriptors` and add a boolean input `extract_keys: bool`. This way the caller would know that this method is intended to check descriptors. However, if they also wish to extract keys here, they can.

Reply:
We will have to change the signature of `LoadParams` though.

```
pub struct LoadParams {
    pub(crate) check_descriptor: Option<(DescriptorToExtract, bool)>,
    pub(crate) check_change_descriptor: Option<(DescriptorToExtract, bool)>,
}
```

where the `bool` is whether to extract keys for signers.

Reply:
I'm actually fine with the current behavior to automatically extract keys. One thing that's still unclear to me is why we have both `descriptors` and `keymap` options

Reply:
@ValuedMammal

In terms of why `keymap` is there, I'm assuming that most people would store private keys as private keys (instead of as a private-key-descriptor). Or, are you saying, because we already have `Wallet::set_keymap` and associated methods, so we do not need `LoadParams::keymap` method?

In terms of why we want `descriptors` to also extract private keys, is because, only allowing public-key-descriptor input is deemed too "breaking" for our users (since they can input a secret-key-descriptor while loading before).

Reply:
yes referring to both methods on `LoadParams` specifically, considering they do nearly the same thing. I wondered if there was a deeper reason - like having the ability to sign txs with keys that don't actually belong to the wallet.

Reply:
That's an interesting though, I don't think that would be an intended feature.

**Here are my current thoughts on the topic:**

We need a `keymap` method on `LoadParams`, otherwise people who store secrets as raw keys need to re-craft secret-key descriptor(s) from their keys (which makes a bad API).

We need a `descriptor` method because we want the caller to be able to check the descriptor against what is loaded.

I also think the main intention of the `descriptor` method should be to check the loaded descriptor. Extracting the secret keys should be a "side quest". Therefore, I think my original proposal makes sense (to rename to `check_descriptor` and have a boolean to `extract_secrets`).

Reply:
In this scenario would it be an error to pass in `true` for extract-keys if the descriptor only has an xpub?

Reply:
@ValuedMammal that's a good point. I think it's okay just to ignore that fact? "There just aren't any keys to extract, so we do nothing" seems sound?